src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs

```diff
@@ -260,7 +217,7 @@ async Task Process(BasicDeliverEventArgs message, byte[] messageBody)
                         var contextBag = new ContextBag();
                         contextBag.Set(message);
 
-                        var messageContext = new MessageContext(messageId, headers, messageBody ?? Array.Empty<byte>(), transportTransaction, tokenSource, contextBag);
+                        var messageContext = new MessageContext(messageId, headers, message.Body.ToArray() ?? Array.Empty<byte>(), transportTransaction, tokenSource, contextBag);
```

Comment:
Do we want to recreate the array every time? Seems like we could do this once earlier in the method and then reuse the result.

Reply:
What if a mutator changes the byte array during the execution and then if the execution fails those changes would leak into the error pipeline or potentially worse the FLRs right?

Reply:
This would mean that the mutator has changed individual elements inside the array, which seems unlikely. That most likely result in message body that couldn't be deserialized.

If you look back to the message pump before we reacted to the 6.0 changes, [it just directly passed `message.Body` into the Context ctors](https://github.com/Particular/NServiceBus.RabbitMQ/blob/5.2.0/src/NServiceBus.Transport.RabbitMQ/Receiving/MessagePump.cs#L251), so I don't think there would be any harm in going back to something similar by doing the array conversion once.

Reply:
I'm not a fan of having code that can have side effects across retries but given this was always the case with this transport I'm not opposed to restoring back to this behavior given that you have more context in this transport than I do.

Reply:
> I'm not a fan of having code that can have side effects across retries

In general I'm not either, but I also think that this is core's responsibility not the transport.